Holo-annie-jam/Screens/Gameplay/MainGameScreen.cs

```diff
@@ -160,6 +207,24 @@ public override void Update(GameTime gameTime, bool otherScreenHasFocus,
 
             enemyPosition = Vector2.Lerp(enemyPosition, targetPosition, 0.05f);
         }
+
+        // update visible events and assign quads to visible rhythm events
+        visibleEvents = beatmapPlayer.GetVisibleEvents();
+
+        foreach (RhythmEvent rhythmEvent in visibleEvents.RhythmEvents ?? Array.Empty<RhythmEvent>()) { 
```

Comment:
There has GOT to be a better way to do this, I honestly think it would be more efficient to recalculate the vectors every time than have to keep track of rhythm events that are still visible and update them - alternatively, have GetVisibleEvents() return array indices instead

Reply:
I do this do dodge having to recreate Quad objects on every update loop. Either way we have to recalculate vectors every time, but we save a ton of resources on object creation/destruction on each loop